transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollStreamChannel.java

```diff
@@ -672,4 +836,216 @@ public void run() {
+                // on multiple Channels pointing to one target Channel.
+                int pipeOut = ch.pipeOut;
+                if (pipeOut == -1) {
+                    // Create a new pipe as non was created before.
+                    long fds = Native.pipe();
+                    ch.pipeIn = (int) (fds >> 32);
+                    pipeOut = ch.pipeOut = (int) fds;
+                }
+
+                int splicedIn = spliceIn(pipeOut, handle);
+                if (splicedIn > 0) {
+                    // Integer.MAX_VALUE is a special value which will result in splice forever.
+                    if (len != Integer.MAX_VALUE) {
+                        len -= splicedIn;
+                    }
+
+                    // Depending on if we are done with splicing inbound data we set the right promise for the
+                    // outbound splicing.
+                    ChannelPromise splicePromise;
+                    if (len == 0) {
+                        splicePromise = promise;
+                    } else {
+                        splicePromise = ch.newPromise().addListener(this);
+                    }
+
+                    boolean autoRead = config().isAutoRead();
+
+                    // Just call unsafe().write(...) and flush() as we not want to traverse the whole pipeline for this
+                    // case.
+                    ch.unsafe().write(new SpliceOutTask(ch, splicedIn, autoRead), splicePromise);
+                    ch.unsafe().flush();
+                    if (autoRead && !splicePromise.isDone()) {
+                        // Write was not done which means the target channel was not writable. In this case we need to
+                        // disable reading until we are done with splicing to the target channel because:
+                        //
+                        // - The user may want to to trigger another splice operation once the splicing was complete.
+                        config().setAutoRead(false);
+                    }
+                }
+
+                return len == 0;
+            } catch (Throwable cause) {
+                promise.setFailure(cause);
+                return true;
+            }
+        }
+    }
+
+    private final class SpliceOutTask {
+        private final AbstractEpollStreamChannel ch;
+        private final boolean autoRead;
+        private int len;
+
+        SpliceOutTask(AbstractEpollStreamChannel ch, int len, boolean autoRead) {
+            this.ch = ch;
+            this.len = len;
+            this.autoRead = autoRead;
+        }
+
+        public boolean spliceOut() throws Exception {
+            assert ch.eventLoop().inEventLoop();
+            try {
+                int splicedOut = Native.splice(ch.pipeIn, -1, ch.fd().intValue(), -1, len);
+                len -= splicedOut;
+                if (len == 0) {
+                    if (autoRead) {
```

Comment:
Should this conditional reflect the conditional that was used to set autoread to false in the first place? https://github.com/netty/netty/pull/3665/files#diff-0151984b13fcb94eb32f49bb667a5d29R933